server/commands/subscriptionCreator.ts

```diff
@@ -0,0 +1,70 @@
+import { Transaction } from "sequelize";
+import { Subscription, Event, User } from "@server/models";
+
+type Props = {
+  /** The user creating the subscription */
+  user: User;
+  /** The document to subscription */
+  documentId?: string;
+  /** Event to subscribe */
+  event: string;
+  /** Status of a subscription */
+  enabled: boolean;
+  /** The IP address of the user creating the subscription */
+  ip: string;
+  transaction: Transaction;
+};
+
+/**
+ * This command creates a "subscription" on
+ * a document via the subscription relation.
+ *
+ * @param Props The properties of the subscription to create
+ * @returns Subscription The subscription that was created
+ */
+export default async function subscriptionCreator({
+  user,
+  documentId,
+  event,
+  // Sane default state when creating a subscription.
+  enabled = true,
+  ip,
+  transaction,
+}: Props): Promise<Subscription> {
+  // If a subscription already exists, fetch it,
+  // otherwise create a subscription and return it.
+  // `created` is a boolean value
+  // that indicates  whether
+  // that subscription was created
+  // or already existed.
+  const [subscription, created] = await Subscription.findOrCreate({
+    where: {
+      userId: user.id,
+      documentId,
+      event,
+      enabled,
+    },
+  });
+
+  // Don't emit an event if a new subscription
+  // wasn't created.
```

Comment:
```suggestion
  // Don't emit an event if a new subscription wasn't created.
```

Can you make sure that comments go to the 80 character line wrap, thanks 🙏

Reply:
Will do. My auto-formatter doesn't seem to like eslint/vscode at all so it tends to format things all over the place. I'll do a full sweep after finalizing changes.